data_processing.ipynb

```diff
@@ -748,6 +759,421 @@
+       "51   [0.014029436354815213, 0.014280474969277254, 0...     0\n",
+       "370  [0.03026395156144915, 0.030441104676146632, 0....     1\n",
+       "456  [0.018051205585293762, 0.020033789914877715, 0...     2\n",
+       "102  [0.025893012259715704, 0.021900611378626946, 0...     0\n",
+       "36   [0.019407836535601047, 0.0187561280211469, 0.0...     0\n",
+       "..                                                 ...   ...\n",
+       "8    [0.02746254970056796, 0.022406735048318086, 0....     0\n",
+       "107  [0.02350585332112601, 0.025473906998611087, 0....     2\n",
+       "490  [0.02423272928835719, 0.02342685552305181, 0.0...     2\n",
+       "28   [0.015755475703704988, 0.01818003315805917, 0....     3\n",
+       "4    [0.02230706281282789, 0.021776700824309162, 0....     0\n",
+       "\n",
+       "[153 rows x 2 columns]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "3    53\n",
+      "0    37\n",
+      "2    33\n",
+      "1    30\n",
+      "Name: label, dtype: int64\n"
+     ]
+    }
+   ],
+   "source": [
+    "def split_GPAC_data():\n",
+    "    \"\"\"\n",
+    "    Function that takes all of the data, filters out the rows w/ a label of -1, and then splits \n",
+    "    the data into Training, Validation, and Test \n",
+    "    \"\"\"\n",
+    "    \n",
+    "    all_data = pd.DataFrame(data=None, columns=['fv', 'label'])\n",
+    "    for dataset in gpac_image_data:\n",
+    "        \n",
+    "        # Isolate the image data portion of each dataset\n",
+    "        sub_data = gpac_image_data[dataset]['data']\n",
+    "        sub_data = sub_data.drop(['window_start'], axis=1)\n",
+    "        \n",
+    "        # Now we need to filter out the rows where the 'label' value is -1\n",
+    "        df1 = sub_data[sub_data['label'] > -1]\n",
+    "        all_data = all_data.append(df1)\n",
+    " \n",
+    "    # Now is the splitting of the dataframe into train, test, and validation datafames\n",
+    "    train_data, valid_data, test_data = np.split(all_data.sample(frac=1, random_state=42),\n",
+    "                                                [int(0.7 * len(all_data)), int(0.85 * len(all_data))])\n",
+    "\n",
+    "    return train_data, valid_data, test_data\n",
+    "\n",
+    "train_GPAC_data, valid_GPAC_data, test_GPAC_data = split_GPAC_data()\n",
+    "\n",
+    "print('\\n\\n\\nTraining DataFrame and class breakdown')\n",
+    "display(train_GPAC_data)\n",
+    "print(train_GPAC_data['label'].value_counts())\n",
+    "\n",
+    "print('\\n\\n\\nValidation DataFrame and class breakdown')\n",
+    "display(valid_GPAC_data)\n",
+    "print(valid_GPAC_data['label'].value_counts())\n",
+    "\n",
+    "\n",
+    "print('\\n\\n\\nTesting DataFrame and class breakdown')\n",
+    "display(test_GPAC_data)\n",
+    "print(test_GPAC_data['label'].value_counts())"
```

Comment:
I'm kind of confused here. In the previous cell you printed out label counts for the GPAC data, where there were 15 of each label:
![image](https://user-images.githubusercontent.com/34193060/100529667-db1fc280-31b7-11eb-8af7-32e0a5bec623.png)

However, here, you're splitting the GPAC data and label 1, for example, has 300 samples spread across the three different sets. Is there something I'm missing here? Shouldn't these be equal?

Reply:
In the previous cell, that was only with 1 dataset ('data_AL_D6_2_1_export.mat').

In the cell w/ the splitting, I'm using all of the data (all of the .mat files)

Reply:
Yeah we definitely need to find more data then, lol.